shiny/ui/_markdown_stream.py

```diff
@@ -0,0 +1,276 @@
+          otherwise display the actual error message.
+        * `"actual"`: Display the actual error message to the user.
+        * `"sanitize"`: Sanitize the error message before displaying it to the user.
+        * `"unhandled"`: Do not display any error message to the user.
+
+    Note
+    ----
+    Markdown is parsed on the client via `marked.js`. Consider using :func:`~shiny.ui.markdown`
+    for server-side rendering of markdown content.
+    """
+
+    def __init__(
+        self,
+        id: str,
+        *,
+        on_error: Literal["auto", "actual", "sanitize", "unhandled"] = "auto",
+    ):
+        self.id = id
+        # TODO: remove the `None` when this PR lands:
+        # https://github.com/posit-dev/py-shiny/pull/793/files
+        self._session = require_active_session(None)
+
+        # Default to sanitizing until we know the app isn't sanitizing errors
+        if on_error == "auto":
+            on_error = "sanitize"
+            app = self._session.app
+            if app is not None and not app.sanitize_errors:  # type: ignore
+                on_error = "actual"
+
+        self.on_error = on_error
+
+    def ui(
+        self,
+        *,
+        content: str = "",
+        content_type: StreamingContentType = "markdown",
+        auto_scroll: bool = True,
+        width: CssUnit = "100%",
+        height: CssUnit = "auto",
+    ) -> Tag:
+        """
+        Get the UI element for this markdown stream.
+
+        This method is only relevant for Shiny Express. In Shiny Core, use
+        :func:`~shiny.ui.output_markdown_stream` for placing the markdown stream
+        in the UI.
+
+        Parameters
+        ----------
+        content
+            Some content to display before any streaming occurs.
+        content_type
+            The content type. Default is "markdown" (specifically, CommonMark).
+            Other supported options are:
+            - `"html"`: for rendering HTML content.
+            - `"text"`: for plain text.
+            - `"semi-markdown"`: for rendering markdown, but with HTML tags escaped.
+        auto_scroll
+            Whether to automatically scroll to the bottom of a scrollable container
+            when new content is added. Default is True.
+        width
+            The width of the markdown stream container.
+        height
+            The height of the markdown stream container.
+
+        Returns
+        -------
+        Tag
+            The UI element for this markdown stream.
+        """
+        return markdown_stream_ui(
+            self.id,
+            content=content,
+            content_type=content_type,
+            auto_scroll=auto_scroll,
+            width=width,
+            height=height,
+        )
+
+    async def stream(
+        self,
+        content: Iterable[str] | AsyncIterable[str],
+        clear: bool = True,
+    ):
+        """
+        Stream content into the markdown stream.
```

Comment:
To touch on "markdown stream" one more time, I don't think it's a great term to describe the UI or component that's in the app. One option could be to just capitalize "Markdown Stream". Otherwise, markdown stream here seems ephemeral and it's not immediately obvious that this means "stream into the content into the UI".

Reply:
I'm going to stop calling this out; I think it's worth doing a quick review of all uses.

Reply:
Addressed in [d091ebd](https://github.com/posit-dev/py-shiny/pull/1782/commits/d091ebdedb97b6f222f8284c5ac05695d1942b42)